File: javascript/octal/octal.js
var Octal = function (octal) {
  this.octal = octal;
};

Octal.prototype.toDecimal = function () {
  var result = 0;
  for (var i = 0; i < this.octal.length; i++) {
    if (this.octal[i] >= '0' && this.octal[i] <= '7') {
      result += this.octal[i] * Math.pow(8, this.octal.length - (i + 1));
    } else {
      return 0;
    }
  }
  return result;
};

module.exports = Octal;

File: csharp/octal/Octal.cs
using System;

class Octal
{
  public static int ToDecimal(string value)
  {
    var result = 0;
    var position = value.Length - 1;
    foreach (var c in value)
    {
      if (char.IsDigit(c))
      {
        var number = (int)char.GetNumericValue(c);
        if (number > 7)
          return 0;
        result += number * (int)Math.Pow(8, position);
        position--;
      }
      else
      {
        return 0;
      }
    }
    return result;
  }
}